packages/cli/src/components/ProjectStatus/ServerProjectStatusRows.tsx

```diff
@@ -0,0 +1,166 @@
+          <Text>{url} </Text>
+          <Text color="green">✓ </Text>
+          <Text dimColor>{event.ms}ms</Text>
+        </Box>
+      )}
+
+      {event.type === "RPC_FAILED_NOT_FOUND_ERROR" && (
+        <Box flexDirection="column">
+          <Box>
+            <Text dimColor>{event.dateTime} </Text>
+            <Text>{url} </Text>
+            <Text color="red">✖ </Text>
+            <Text dimColor>{event.ms}ms</Text>
+          </Box>
+
+          <Box marginX={2} marginY={1}>
+            <Text color="red">Function not found</Text>
+          </Box>
+        </Box>
+      )}
+
+      {event.type === "RPC_FAILED_VALIDATION_ERROR" && (
+        <Box flexDirection="column">
+          <Box>
+            <Text dimColor>{event.dateTime} </Text>
+            <Text>{url} </Text>
+            <Text color="red">✖ </Text>
+            <Text dimColor>{event.ms}ms</Text>
+          </Box>
+
+          <Box marginX={2} marginY={1} flexDirection="column">
+            <Box marginBottom={1}>
+              <Text color="red">
+                {event.errors.length === 1
+                  ? "Validation error"
+                  : "Validation errors"}
+              </Text>
+            </Box>
+
+            <KeyValueTable
+              content={event.errors.reduce(
+                (result, error) => ({
+                  ...result,
+                  [error.path]: error.message,
+                }),
+                {},
+              )}
+            />
+          </Box>
+        </Box>
+      )}
+
+      {event.type === "RPC_FAILED_FUNCTION_ERROR" && (
+        <Box flexDirection="column">
+          <Box>
+            <Text dimColor>{event.dateTime} </Text>
+            <Text>{url} </Text>
+            <Text color="red">✖ </Text>
+            <Text dimColor>{event.ms}ms</Text>
+          </Box>
+
+          <Box marginX={2} marginY={1} flexDirection="column">
+            <Box marginBottom={1}>
+              <Text color="red">{event.error.name}</Text>
+            </Box>
+
+            <Box marginBottom={1}>
+              <KeyValueTable content={event.error.props} />
+            </Box>
+
+            <Box>
+              <Text>{event.error.stack}</Text>
+            </Box>
+          </Box>
+        </Box>
+      )}
+
+      {event.type === "RPC_FAILED_SERVER_ERROR" && (
+        <Box flexDirection="column">
+          <Box>
+            <Text dimColor>{event.dateTime} </Text>
+            <Text>{url} </Text>
+            <Text color="red">✖ </Text>
+            <Text dimColor>{event.ms}ms</Text>
+          </Box>
+
+          <Box marginX={2} marginY={1} flexDirection="column">
+            <Text>{event.error.message}</Text>
+            <Text>{event.error.stack}</Text>
+          </Box>
+        </Box>
+      )}
+    </>
+  )
+}
+
+function KeyValueTable({ content }: { content: Record<string, any> }) {
+  const entries = Object.entries(content)
+  const maxLabelLength = maxLength(entries.map(([k]) => k))
+  const format = (value: any) => {
```

Comment:
Isn't this the same as:
`const format = (value: any) => typeof value === "undefined" ? "undefined" : JSON.stringify(value)`
?